test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs

```diff
@@ -257,6 +259,36 @@ public TestHostTest()
             Assert.Equal("TestExecution.Response", host.Output[host.Output.Count - 1].MessageType);
         }
 
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public async Task TestHostExits_WhenParentProcessExits()
+        {
+            // Arrange
+            var parentProcess = new Process();
+            parentProcess.StartInfo.FileName = "cmd";
+            parentProcess.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
+            var testHost = new TestHostWrapper(_testProject);
+            int testHostProcessId;
+
+            try
+            {
+                parentProcess.Start();
+                testHost.ParentProcessId = parentProcess.Id;
+
+                // Act
+                await testHost.StartAsync();
+                testHostProcessId = testHost.Process.Id;
+            }
+            finally
+            {
+                parentProcess.Kill();
+            }
+
+            // Assert
+            await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
A better way to do this would be to wait *up to* 5 seconds.

If you find the testHostProcess attach to it and wait for **it** to exit with a timeout.